cli_client/python/timesketch_cli_client/commands/search.py

```diff
@@ -0,0 +1,210 @@
+@click.option(
+    '--limit', type=int, default=40,
+    help='Limit amount of events to show (default: 40)')
+@click.option(
+    '--saved-search', type=int, help='Query and filter from saved search')
+@click.option(
+    '--describe', is_flag=True, default=False,
+    help='Show the query and filter then exit')
+@click.pass_context
+# pylint: disable=too-many-arguments
+def search_group(ctx, query, times, time_ranges, labels, header, output,
+                 return_fields, order, limit, saved_search, describe):
+    """Search and explore."""
+    sketch = ctx.obj.sketch
+    output_format = ctx.obj.output_format
+    search_obj = search.Search(sketch=sketch)
+
+    if output:
+        output_format = output
+
+    new_line = True
+    if output_format == 'csv':
+        new_line = False
+
+    # Construct query from saved search and return early.
+    if saved_search:
+        search_obj.from_saved(saved_search)
+        if describe:
+            describe_query(search_obj)
+            return
+        click.echo(format_output(
+            search_obj, output_format, header), nl=new_line)
+        return
+
+    # Construct the query from flags.
+    # TODO (berggren): Add support for query DSL.
+    search_obj.query_string = query
+    search_obj.return_fields = return_fields
+    search_obj.max_entries = limit
+
+    if order == 'asc':
+        search_obj.order_ascending()
+    elif order == 'desc':
+        search_obj.order_descending()
+
+    # TODO: Add term chips.
+    if time_ranges:
+        for time_range in time_ranges:
+            range_chip = search.DateRangeChip()
+            range_chip.start_time = time_range[0]
+            range_chip.end_time = time_range[1]
+            search_obj.add_chip(range_chip)
+
+    # TODO (berggren): This should support dates like 2021-02-12 and then
+    # convert to ISO format.
+    if times:
+        for time in times:
+            range_chip = search.DateRangeChip()
+            range_chip.start_time = time
+            range_chip.end_time = time
+            search_obj.add_chip(range_chip)
+
+    if labels:
+        for label in labels:
+            label_chip = search.LabelChip()
+            if label == 'star':
+                label_chip.use_star_label()
+            elif label == 'comment':
+                label_chip.use_comment_label()
+            else:
+                label_chip.label = label
+            search_obj.add_chip(label_chip)
+
+    if describe:
+        describe_query(search_obj)
+        return
+
+    click.echo(format_output(search_obj, output_format, header), nl=new_line)
+
+
+@click.group('saved-searches')
+def saved_searches_group():
+    """Managed saved searches."""
+
+
+@saved_searches_group.command('list')
+@click.pass_context
+def list_saved_searches(ctx):
+    """List saved searches in the sketch.
+
+    Args:
+        ctx: Click CLI context object.
+    """
+    sketch = ctx.obj.sketch
+    for saved_search in sketch.list_views():
```

Comment:
let's not use this function but rather `sketch.list_saved_searches` since that is the new one, the `list_views` will be deprecated soon

Reply:
Done